src/canvascontext.ts

```diff
@@ -221,6 +221,26 @@ export class CanvasContext implements RenderContext {
     return this;
   }
 
+  /**
+   * @param path a SVG path string
+   */
+  fillPath(path: string, scale: number, x: number, y: number): this {
+    const ctx = this.vexFlowCanvasContext;
+    ctx.save();
```

Comment:
Great suggestion! We definitely should opt for lighter weight alternatives when possible. I was just going by the VexFlow code convention (since I've only been a contributor for about 6 months).

I see a bunch of places (30+) where we use ctx.save() and ctx.restore(). Perhaps we can optimize all of these for ver 4.1.

https://github.com/0xfe/vexflow/blob/660f488b33e71e8a3ff1484c448e2efeaddfc4fa/src/stavevolta.ts#L74-L77

https://github.com/0xfe/vexflow/blob/660f488b33e71e8a3ff1484c448e2efeaddfc4fa/src/strokes.ts#L247-L250